tests/transform.js

```diff
@@ -85,4 +92,177 @@ describe('Transform', () => {
         assert(options.treeAdapter instanceof Object);
         assert.equal(options.handleTags, handleTags);
     });
+
+    it('should return correct serialized chunks', () => {
```

Comment:
Not sure that we need to stub serializer here. I would prefer to pass sinon.spy instead and check that it returned the expected result.

Reply:
I also think that the approach you took to implement ignore tags is wrong because it complicates testing. I think it should be implemented as a separate class that will be created on per-request basis and will perform only extraction & insertion of the chunks. So it will be easy to cover it with tests. 

Only be aware that all the regular expressions should be created only once.